vulcan/lib/orchestration.rb

```diff
@@ -57,9 +57,11 @@ def load_json_inputs!(storage)
 
       # Ensure that the primary inputs, at least, have been loaded so that the status endpoint can report
       # meaningful steps running.
-      pi = build_target_for(:primary_inputs)
-      if pi.should_build?(storage)
-        run!(storage: storage, build_target: pi, token: nil)
+      workflow.inputs.each do |input|
+        pi = build_target_for(workflow.step_key(input))
+        if pi.should_build?(storage)
+          run!(storage: storage, build_target: pi, token: nil)
```

Comment:
This is probably the performance hit that causes the commit and run to perform slow.
Is it possible to change run! to execute multiple primary input file loads together?  Or at least share some of the overhead at a lower cost?